sklearn/utils/estimator_checks.py

```diff
@@ -373,7 +378,23 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
-def parametrize_with_checks(estimators):
+def _get_xfail_checks(estimator, strict_mode):
+    # Return the checks that are in the estimator's _xfail_checks tag, along
+    # with the strict checks if strict_mode is False.
+    xfail_checks = estimator._get_tags()['_xfail_checks'] or {}
+
+    if not strict_mode:
+        strict_checks = {
+            _set_check_estimator_ids(check):
+            'The check is strict and strict mode is off'  # the reason
+            for check in _STRICT_CHECKS
```

Comment:
I think it might be  earlier to maintain if it was a decorator on the checks rather than a global list of functions. Didn't we try that in one of the N earlier versions ? :)

Reply:
> Didn't we try that in one of the N earlier versions

One version required a global var, and another one required to pass a `strict` param to every single check, so they're less ideal.

I guess we could have a `@is_strict` decorator that would set an attribute on the check, but it seems a bit overkill?

Reply:
OK, let's start with the current approach, we can always add decorators later.